src/server/session.ts

```diff
@@ -691,13 +690,30 @@ namespace ts.server {
                 };
             }
 
-            return definitionAndBoundSpan;
+            return {
+                ...definitionAndBoundSpan,
+                definitions: definitionAndBoundSpan.definitions.map(Session.mapToOriginalLocation)
+            };
         }
 
         private mapDefinitionInfo(definitions: ReadonlyArray<DefinitionInfo>, project: Project): ReadonlyArray<protocol.FileSpan> {
             return definitions.map(def => this.toFileSpan(def.fileName, def.textSpan, project));
         }
 
+        private static mapToOriginalLocation<T extends DocumentSpan>(def: T): T {

```

Comment:
>mapToOriginalLocation [](start = 23, length = 21)

This name isn't very descriptive but I can't think of one that would be.  I'd suggest putting a simplified version of your commit message in the doc comment for this function.